### PR TITLE
Add onItemHighlighted prop

### PR DESCRIPTION
**Pull request by SamuelBolduc**
Useful for updating other parts of the page in reaction to the highlighted item in the autocomplete dropdown being changed.

In my case I want to reference rows in a table by their name, and when a row's name is highlighted in the autocomplete dropdown I want to highlight the actual row. This PR aims to enable this functionality.

If the name `onItemHighlighted` doesn't make sense, let me know and we can find a better one, I'll update it.

I also bumped the version to v4.7.0 (minor version bump) since this is more than just a bugfix. Again, I can revert this if you have another methodology to manage versions.

### CODE REVIEW

**Comment by jukben** [comment]
Hey! Thank you so much for your PR!

[Here is the playground for this revision](https://1477-94480675-gh.circle-artifacts.com/0/example/index.html) 🚀

I hope everything is fine! ❤️

_This message has been generated [automagically](https://github.com/jukben/rta-bot) ✨_

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@webscopeio/react-textarea-autocomplete",
-  "version": "4.6.3",
+  "version": "4.7.0",
```
Comment (jukben) [inline package.json:3]: Can you please remove this change? I will do it during release process. I agree it should be 4.7.0 🙏

**Comment by SamuelBolduc** [comment]
Alright, makes sense. I reverted it back to `4.6.3` as it was before the commit. Thanks!

**Comment by jukben** [comment]
Released https://github.com/webscopeio/react-textarea-autocomplete/releases/tag/v4.7.0 🚀

**Pull request merged**